R/update-national.R

```diff
@@ -0,0 +1,85 @@
+# Packages -----------------------------------------------------------------
+require(EpiNow2, quietly = TRUE)
+require(covidregionaldata, quietly = TRUE)
+require(data.table, quietly = TRUE)
+require(future, quietly = TRUE)
+require(here, quietly = TRUE)
+require(lubridate, quietly = TRUE)
+require(futile.logger, quietly = TRUE)
+
+# Load utils --------------------------------------------------------------
+
+source(here::here("R", "utils.R"))
+
+options <- c("cases", "deaths")
+
+generation_time <- readRDS(here::here("data", "generation_time.rds"))
+incubation_period <- readRDS(here::here("data", "incubation_period.rds"))
+
+for (option in options) {
+
+  # Update delays -----------------------------------------------------------
+
+  if (option == "cases") {
+    reporting_delay <- readRDS(here::here("data", "onset_to_admission_delay.rds"))
+  }
+  if (option == "deaths") {
+    reporting_delay <- readRDS(here::here("data", "onset_to_death_delay.rds"))
+  }
+
+  # Set up logging ----------------------------------------------------------
+
+  setup_log()
+
+  futile.logger::flog.info("Processing national dataset for: %s", option)
+
+  # Get cases  ---------------------------------------------------------------
+
+  cases <- data.table::setDT(covidregionaldata::get_national_data(source = "ecdc"))
```

Comment:
This gives me some concern as if the deaths data fails to be pulled down I think cases may default to the first iteration of the for loop. That may, in turn, lead to some strange behaviour (that isn't just an error).

Reply:
Most of these issues can probably be solved by making an update national generic function and then operating over that using `lapply` (as the cases variable used in the function won't be inherited by the global environment). Sure there are other solutions/error catches that would work as well.